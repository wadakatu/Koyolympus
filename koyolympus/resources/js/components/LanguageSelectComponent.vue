<template>
    <ul class="language">
        <li><a>Language</a>
            <ul>
                <li><a @click="$router.push('/aboutme')">日本語</a></li>
                <li><a @click="$router.push('/aboutme/english')">English</a></li>
                <li><a>French</a></li>
                <li><a>한국말</a></li>
                <li><a>中文</a></li>
            </ul>
        </li>
    </ul>
</template>

<script>
export default {
    name: "LanguageSelectComponent.vue"
}
</script>

<style scoped>

.language {
    margin-left: 85%;
    background-color: #1a5293;
    color: #fff;
    position: relative;
    list-style-type: none;
    width: 80px;
    height: 25px;
    border-radius: 15px;
    padding-left: 8px;
}

.language li {
    width: 25%;
    margin: 0;
    padding: 0;
    text-align: center;
}

.language ul a {
    display: block;
    margin: 0;
    padding: 13px 0;
    color: #fff;
    font-size: 16px;
    font-weight: bold;
    line-height: 1;
    text-decoration: none;
}

.language li ul {
    list-style: none;
    position: absolute;
    top: 100%;
    left: 0;
    margin: 0;
    padding: 0;
}

.language li ul li {
    overflow: hidden;
    width: 80px;
    height: 0;
    color: #fff;
    -moz-transition: .2s;
    -webkit-transition: .2s;
    -o-transition: .2s;
    -ms-transition: .2s;
    transition: .2s;
}

.language li ul li a {
    padding: 13px 15px;
    background: #1a5293;
    text-align: left;
    font-size: 12px;
    font-weight: normal;
}

.language li:hover > a {
    background: #1a5293;
    color: #e0dc62;
}

.language li:hover ul li {
    overflow: visible;
    height: 38px;
    border-bottom: 1px solid #fff;
}

.language li:hover ul li:first-child {
    border-top: 0;
}

.language li:hover ul li:last-child {
    border-bottom: 0;
}

</style>
