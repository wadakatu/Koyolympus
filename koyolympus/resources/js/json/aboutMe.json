{
    "japanese": [
        "名前",
        "磯野 公耀",
        "生息地",
        "東京都と埼玉県の狭間",
        "年齢/星座/誕生日",
        "22歳/おひつじ座/3月30日",
        "趣味",
        "カメラ/ロードバイク/ゲーム",
        "私のカメラ",
        "OLYMPUS PEN-F/OMD EM-1 MarkⅡ",
        "座右の銘",
        "Fake it until you make it."
    ],
    "english": [
        "My Name",
        "Koyo Isono",
        "My Address",
        "Japan, between Tokyo and Saitama Pref.",
        "My Age/Zodiac sign/Birthday",
        "22-year-old / Aries / March 30th",
        "My interest",
        "Camera / Road Bike / Video game",
        "My Camera",
        "OLYMPUS PEN-F / OMD EM-1 MarkⅡ",
        "My Motto",
        "Fake it until you make it."
    ],
    "french": [
        "Mon Nom",
        "Koyo Isono",
        "Mon Adresse",
        "Japan, entre Tokyo et Saitama.",
        "Mon Âge/Signe du zodiaque/Anniversaire",
        "22 ans / Le Bélier / Le 30 Mars",
        "Mes centres d’intérêt",
        "Caméra / Vélo de route / Jeu vidéo",
        "Ma caméra",
        "OLYMPUS PEN-F / OMD EM-1 MarkⅡ",
        "Ma devise",
        "Tout est dans l'attitude"
    ],
    "korean": [
        "이르",
        "코요 이소노",
        "주소",
        "일본, 도쿄와 사이타마 현 사이",
        "나이/ 별자리/ 생일",
        "22살 / 양자리 / 3월 30일",
        "취미",
        "사진 찍기 / 로드 바이킹 / 비디오 게임",
        "내 카메라",
        "올림푸스 PEN-F/ OMD EM-1 MarkⅡ",
        "내 좌우명",
        "될 때까지 부딫혀보자."
    ],
    "chinese": [
        "我的名字",
        "磯野公耀",
        "我的地址",
        "日本东京 埼玉县",
        "年龄/标志/生日",
        "22岁/白羊座/三月三十号",
        "我的爱好",
        "摄影/公路车/电竞游戏",
        "我的相机",
        "OLYMPUS PEN-F/OMD EM-1 MarkⅡ",
        "我的格言",
        "Fake it until you make it."
    ]
}
