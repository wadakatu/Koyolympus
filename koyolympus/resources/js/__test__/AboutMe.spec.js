import {shallowMount} from '@vue/test-utils';
import Component from '@/AboutMeComponent.vue';
import aboutMe from '../json/aboutMe.json';
import LanguageSelectComponent from "../components/LanguageSelectComponent";

let wrapper;

beforeEach(() => {
    wrapper = shallowMount(Component)
})

describe('Testing data', () => {
    test('language has aboutMe.japanese', () => {
        expect(wrapper.vm.language).toBe(aboutMe.japanese);
    });
    test('question has string which is `question`', () => {
        expect(wrapper.vm.question).toBe('question');
    });
    test('answer has string which is `answer`', () => {
        expect(wrapper.vm.answer).toBe('answer');
    });
});

describe('Testing methods', () => {
    describe('updateLang function', () => {
        test('Japanese ver', () => {
            wrapper.vm.updateLang('japanese');
            expect(wrapper.vm.language.length).toBe(12);
            expect(wrapper.vm.language).toBe(aboutMe.japanese);
        });
        test('English ver', () => {
            wrapper.vm.updateLang('english');
            expect(wrapper.vm.language.length).toBe(12);
            expect(wrapper.vm.language).toBe(aboutMe.english);
        });
        test('French ver', () => {
            wrapper.vm.updateLang('french');
            expect(wrapper.vm.language.length).toBe(12);
            expect(wrapper.vm.language).toBe(aboutMe.french);
        });
        test('Korean ver', () => {
            wrapper.vm.updateLang('korean');
            expect(wrapper.vm.language.length).toBe(12);
            expect(wrapper.vm.language).toBe(aboutMe.korean);
        });
        test('Chinese ver', () => {
            wrapper.vm.updateLang('chinese');
            expect(wrapper.vm.language.length).toBe(12);
            expect(wrapper.vm.language).toBe(aboutMe.chinese);
        });
    });
});

describe('Testing top button', () => {
    test('never click top button', () => {
        const routerPush = jest.fn();
        wrapper = shallowMount(Component, {
            mocks: {
                $router: {
                    push: routerPush
                },
            }
        });
        expect(routerPush.mock.calls.length).toBe(0);
    });
    test('click top button once', () => {
        const routerPush = jest.fn();
        wrapper = shallowMount(Component, {
            mocks: {
                $router: {
                    push: routerPush
                }
            }
        });
        wrapper.find('button.top_button').trigger('click');
        expect(routerPush.mock.calls.length).toBe(1);
        expect(routerPush).toHaveBeenCalledWith('/');
    });
});

describe('Testing language select component', () => {
    test('called updateLang function', () => {
        const updateLangSpy = jest.spyOn(Component.methods, 'updateLang')
        wrapper = shallowMount(Component);
        wrapper.findComponent(LanguageSelectComponent).vm.$emit('update')
        expect(updateLangSpy).toHaveBeenCalled()
    });
});

describe('Testing f-for', () => {
    test('rendered self-introduction correctly', () => {
        const intro = wrapper.findAll('p');
        expect(intro.at(0).classes()).toContain('question');
        expect(intro.at(1).classes()).toContain('answer');
        expect(intro.at(10).classes()).toContain('question');
        expect(intro.at(11).classes()).toContain('answer');
    });
});

describe('Testing snapshot', () => {
    test('default [japanese introduction]', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    test('english introduction', async () => {
        await wrapper.vm.updateLang('english');
        expect(wrapper.element).toMatchSnapshot();
    });
    test('french introduction', async () => {
        await wrapper.vm.updateLang('french');
        expect(wrapper.element).toMatchSnapshot();
    });
    test('korean introduction', async () => {
        await wrapper.vm.updateLang('korean');
        expect(wrapper.element).toMatchSnapshot();
    });
    test('chinese introduction', async () => {
        await wrapper.vm.updateLang('chinese');
        expect(wrapper.element).toMatchSnapshot();
    });
});
