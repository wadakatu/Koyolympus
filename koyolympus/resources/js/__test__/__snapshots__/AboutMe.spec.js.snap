// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing snapshot chinese introduction 1`] = `
<div
  class="col"
>
  <h1
    class="aboutMe"
  >
    About me
  </h1>
   
  <ul
    class="language"
    ontouchstart=""
  >
    <li
      class="language_title"
    >
      <a>
        Language
      </a>
       
      <ul
        class="language_ul"
      >
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            日本語
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            English
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            French
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            한국말
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            中文
          </a>
        </li>
      </ul>
    </li>
  </ul>
   
  <div
    class="intro"
  >
    <p
      class="question"
    >
      
            我的名字
        
    </p>
    <p
      class="answer"
    >
      
            磯野公耀
        
    </p>
    <p
      class="question"
    >
      
            我的地址
        
    </p>
    <p
      class="answer"
    >
      
            日本东京 埼玉县
        
    </p>
    <p
      class="question"
    >
      
            年龄/标志/生日
        
    </p>
    <p
      class="answer"
    >
      
            22岁/白羊座/三月三十号
        
    </p>
    <p
      class="question"
    >
      
            我的爱好
        
    </p>
    <p
      class="answer"
    >
      
            摄影/公路车/电竞游戏
        
    </p>
    <p
      class="question"
    >
      
            我的相机
        
    </p>
    <p
      class="answer"
    >
      
            OLYMPUS PEN-F/OMD EM-1 MarkⅡ
        
    </p>
    <p
      class="question"
    >
      
            我的格言
        
    </p>
    <p
      class="answer"
    >
      
            Fake it until you make it.
        
    </p>
  </div>
   
  <button
    class="top_button"
  >
    TOPへ
  </button>
</div>
`;

exports[`Testing snapshot default [japanese introduction] 1`] = `
<div
  class="col"
>
  <h1
    class="aboutMe"
  >
    About me
  </h1>
   
  <ul
    class="language"
    ontouchstart=""
  >
    <li
      class="language_title"
    >
      <a>
        Language
      </a>
       
      <ul
        class="language_ul"
      >
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            日本語
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            English
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            French
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            한국말
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            中文
          </a>
        </li>
      </ul>
    </li>
  </ul>
   
  <div
    class="intro"
  >
    <p
      class="question"
    >
      
            名前
        
    </p>
    <p
      class="answer"
    >
      
            磯野 公耀
        
    </p>
    <p
      class="question"
    >
      
            生息地
        
    </p>
    <p
      class="answer"
    >
      
            東京都と埼玉県の狭間
        
    </p>
    <p
      class="question"
    >
      
            年齢/星座/誕生日
        
    </p>
    <p
      class="answer"
    >
      
            22歳/おひつじ座/3月30日
        
    </p>
    <p
      class="question"
    >
      
            趣味
        
    </p>
    <p
      class="answer"
    >
      
            カメラ/ロードバイク/ゲーム
        
    </p>
    <p
      class="question"
    >
      
            私のカメラ
        
    </p>
    <p
      class="answer"
    >
      
            OLYMPUS PEN-F/OMD EM-1 MarkⅡ
        
    </p>
    <p
      class="question"
    >
      
            座右の銘
        
    </p>
    <p
      class="answer"
    >
      
            Fake it until you make it.
        
    </p>
  </div>
   
  <button
    class="top_button"
  >
    TOPへ
  </button>
</div>
`;

exports[`Testing snapshot english introduction 1`] = `
<div
  class="col"
>
  <h1
    class="aboutMe"
  >
    About me
  </h1>
   
  <ul
    class="language"
    ontouchstart=""
  >
    <li
      class="language_title"
    >
      <a>
        Language
      </a>
       
      <ul
        class="language_ul"
      >
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            日本語
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            English
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            French
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            한국말
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            中文
          </a>
        </li>
      </ul>
    </li>
  </ul>
   
  <div
    class="intro"
  >
    <p
      class="question"
    >
      
            My Name
        
    </p>
    <p
      class="answer"
    >
      
            Koyo Isono
        
    </p>
    <p
      class="question"
    >
      
            My Address
        
    </p>
    <p
      class="answer"
    >
      
            Japan, between Tokyo and Saitama Pref.
        
    </p>
    <p
      class="question"
    >
      
            My Age/Zodiac sign/Birthday
        
    </p>
    <p
      class="answer"
    >
      
            22-year-old / Aries / March 30th
        
    </p>
    <p
      class="question"
    >
      
            My interest
        
    </p>
    <p
      class="answer"
    >
      
            Camera / Road Bike / Video game
        
    </p>
    <p
      class="question"
    >
      
            My Camera
        
    </p>
    <p
      class="answer"
    >
      
            OLYMPUS PEN-F / OMD EM-1 MarkⅡ
        
    </p>
    <p
      class="question"
    >
      
            My Motto
        
    </p>
    <p
      class="answer"
    >
      
            Fake it until you make it.
        
    </p>
  </div>
   
  <button
    class="top_button"
  >
    TOPへ
  </button>
</div>
`;

exports[`Testing snapshot french introduction 1`] = `
<div
  class="col"
>
  <h1
    class="aboutMe"
  >
    About me
  </h1>
   
  <ul
    class="language"
    ontouchstart=""
  >
    <li
      class="language_title"
    >
      <a>
        Language
      </a>
       
      <ul
        class="language_ul"
      >
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            日本語
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            English
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            French
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            한국말
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            中文
          </a>
        </li>
      </ul>
    </li>
  </ul>
   
  <div
    class="intro"
  >
    <p
      class="question"
    >
      
            Mon Nom
        
    </p>
    <p
      class="answer"
    >
      
            Koyo Isono
        
    </p>
    <p
      class="question"
    >
      
            Mon Adresse
        
    </p>
    <p
      class="answer"
    >
      
            Japan, entre Tokyo et Saitama.
        
    </p>
    <p
      class="question"
    >
      
            Mon Âge/Signe du zodiaque/Anniversaire
        
    </p>
    <p
      class="answer"
    >
      
            22 ans / Le Bélier / Le 30 Mars
        
    </p>
    <p
      class="question"
    >
      
            Mes centres d’intérêt
        
    </p>
    <p
      class="answer"
    >
      
            Caméra / Vélo de route / Jeu vidéo
        
    </p>
    <p
      class="question"
    >
      
            Ma caméra
        
    </p>
    <p
      class="answer"
    >
      
            OLYMPUS PEN-F / OMD EM-1 MarkⅡ
        
    </p>
    <p
      class="question"
    >
      
            Ma devise
        
    </p>
    <p
      class="answer"
    >
      
            Tout est dans l'attitude
        
    </p>
  </div>
   
  <button
    class="top_button"
  >
    TOPへ
  </button>
</div>
`;

exports[`Testing snapshot korean introduction 1`] = `
<div
  class="col"
>
  <h1
    class="aboutMe"
  >
    About me
  </h1>
   
  <ul
    class="language"
    ontouchstart=""
  >
    <li
      class="language_title"
    >
      <a>
        Language
      </a>
       
      <ul
        class="language_ul"
      >
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            日本語
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            English
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            French
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            한국말
          </a>
        </li>
         
        <li
          class="lang"
        >
          <a
            class="lang_detail"
          >
            中文
          </a>
        </li>
      </ul>
    </li>
  </ul>
   
  <div
    class="intro"
  >
    <p
      class="question"
    >
      
            이르
        
    </p>
    <p
      class="answer"
    >
      
            코요 이소노
        
    </p>
    <p
      class="question"
    >
      
            주소
        
    </p>
    <p
      class="answer"
    >
      
            일본, 도쿄와 사이타마 현 사이
        
    </p>
    <p
      class="question"
    >
      
            나이/ 별자리/ 생일
        
    </p>
    <p
      class="answer"
    >
      
            22살 / 양자리 / 3월 30일
        
    </p>
    <p
      class="question"
    >
      
            취미
        
    </p>
    <p
      class="answer"
    >
      
            사진 찍기 / 로드 바이킹 / 비디오 게임
        
    </p>
    <p
      class="question"
    >
      
            내 카메라
        
    </p>
    <p
      class="answer"
    >
      
            올림푸스 PEN-F/ OMD EM-1 MarkⅡ
        
    </p>
    <p
      class="question"
    >
      
            내 좌우명
        
    </p>
    <p
      class="answer"
    >
      
            될 때까지 부딫혀보자.
        
    </p>
  </div>
   
  <button
    class="top_button"
  >
    TOPへ
  </button>
</div>
`;
